tests/integration/test_notification.py

```
@@ -0,0 +1,195 @@
+    ScanCompletedData,
+    PathCreatedData,
+    DiskSpaceCriticalData,
+    ScanErrorData,
+)
+from app.services.notification_service import NotificationService, rate_limiter
+
+# Reset rate limiter for each test
+@pytest.fixture(autouse=True)
+def reset_rate_limiter():
+    rate_limiter.last_notification = {}
+    yield
```

Ruff (0.14.14)

[warning] No teardown in fixture `reset_rate_limiter`, use `return` instead of `yield`

Replace `yield` with `return`

(PT022)